src/Compiler/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/CodeGenerationIntegrationTest.cs

```diff
@@ -6,1096 +6,381 @@
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Linq;
+using System.Runtime.CompilerServices;
 using Microsoft.AspNetCore.Razor.Language.Extensions;
-using Xunit;
+using Microsoft.AspNetCore.Razor.Test.Common;
 
 namespace Microsoft.AspNetCore.Razor.Language.IntegrationTests;
 
-public class CodeGenerationIntegrationTest : IntegrationTestBase
+public class CodeGenerationIntegrationTest(bool designTime = false)
+    : IntegrationTestBase(layer: TestProject.Layer.Compiler, generateBaselines: null)
 {
-    public CodeGenerationIntegrationTest()
-        : base(layer: TestProject.Layer.Compiler, generateBaselines: null)
+    public override string GetTestFileName(string testName)
```

Comment:
Nit: I would prefer all the helpers together; either move this to the bottom of the file with the others, or move the others up here.